modules/new_profile/php/new_profile.class.inc

```diff
@@ -114,28 +114,13 @@ class New_Profile extends \NDB_Form
         if ($request->getMethod() === "POST") {
             $this->save();
             if (!empty($this->_error)) {
-                return (new \LORIS\Http\Response())
-                    ->withHeader("Content-Type", "application/json")
-                    ->withStatus(400)
-                    ->withBody(
-                        new \LORIS\Http\StringStream(
-                            json_encode($this->_error)
-                        )
-                    );
+                return new \LORIS\Http\Response($this->_error);
             }
             $result = array(
                 'candID' => $this->_candID,
                 'pscid'  => $this->_pscid,
             );
-            return (new \LORIS\Http\Response())
-                ->withHeader("Content-Type", "application/json")
-                ->withStatus(201)
-                ->withHeader("Allow", "POST")
-                ->withBody(
-                    new \LORIS\Http\StringStream(
-                        json_encode($result)
-                    )
-                );
+            return new \LORIS\Http\Response\JSON\Created($result);
```

Comment:
This doesn't change the return type either other than using a sub-class of response. It also loses the Allow header.